tapdb/assets_store.go

```diff
@@ -983,6 +984,15 @@ func (a *AssetStore) QueryBalancesByAsset(ctx context.Context,
 		},
 	}
 
+	// We exclude the assets that are specifically used for funding custom
+	// channels. The balance of those assets is reported through lnd channel
+	// balance. Those assets are identified by the funding script tree for a
+	// custom channel asset-level script key.
+	excludeKey := asset.NewScriptKey(
```

Comment:
if I pay assets to an OP_TRUE scriptkey for other purposes (i.e an in-flight trustless swap) it will also ignore them (in that case, it's desired as a pending swap is not part of my balance)

Maybe some other use-case (a not-so-secure wallet..?) is going to hold assets in OP_TRUE script keys and still control the btc level utxos, in that case we would report 0 balance

non-blocking, just worth a mention

Reply:
Well, in the case of a not-so-secure wallet it's also correct behavior, since you should assume it's also not part of your balance anymore. :smile: